src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,347 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.Stream.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP
+namespace Microsoft.OData.Json
+{
+    using System;
+    using System.Buffers;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A stream for writing the stream value.</returns>
+        public Stream StartStreamValueScope()
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            this.WriteItemWithSeparatorIfNeeded();
+
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            this.Flush();
+
+            this.binaryValueStream =  new ODataUtf8JsonWriteStream(this);
+
+            return this.binaryValueStream;
+        }
+
+        /// <summary>
+        /// Ends a scope for writing a stream value.
+        /// </summary>
+        public void EndStreamValueScope()
+        {
+            this.binaryValueStream.Dispose();

```

Comment:
Unless I'm wrong, `this.binaryValueStream` is initialized from either `StartStreamValueScope` or `StartTStreamValueScopeAsync`. Should we check that it's not null from this method just in case someone calls this method without first calling the method that initializes `this.binaryValueStream`?